modules/internal/BytesStringCommon.chpl

```diff
@@ -719,27 +719,45 @@ module BytesStringCommon {
         const localNeedle = needle.localize();
         const needleLen = localNeedle.buffLen;
 
-        // i *is not* an index into anything, it is the order of the element
-        // of view we are searching from.
-        const numPossible = xLen - nLen + 1;
-        const searchSpace = if fromLeft
-            then 0..#(numPossible)
-            else 0..#(numPossible) by -1;
-        for i in searchSpace {
-          // j *is* the index into the localNeedle's buffer
-          for j in 0..#nLen {
-            const idx = view.orderToIndex(i+j); // 0s based idx
-            if x.buff[idx] != localNeedle.buff[j] then break;
-
-            if j == nLen-1 {
-              if count {
-                localRet += 1;
-              } else { // find
-                localRet = view.orderToIndex(i);
-              }
+        const numChar = 256;
+
+        // Preprocessing bad character heurestic
+        var badChar = c_calloc(c_int, numChar);
```

Comment:
A potential performance improvement: somehow retain this allocation somewhere.

malloc is a costly system call, and not having to do this at every call to this function can save time. One idea that I can think of is to make this a module level variable. But then you'd need to mangle the name a bit manually, because `badChar` will be visible in string/bytes implementations, which may not be nice. `chpl_bmh_badChar` or something that starts with `chpl_` should be fine.

Reply:
@e-kayrakli Thanks for the review
> chpl_bmh_badChar or something that starts with chpl_ should be fine.

This seems like a nice idea.